### PR TITLE
SRD support for send WR API

### PR DESCRIPTION
**Pull request by dkkranz**
Introduce post list mode to IBV send WR API, and add support for SRD, including RDMA read.

### CODE REVIEW

File: src/perftest_resources.c [hunk 1]
```diff
@@ -3286,9 +3291,6 @@ void ctx_set_send_reg_wqes(struct pingpong_context *ctx,
 
 			if ((j + 1) % user_param->cq_mod == 0) {
 				ctx->wr[i*user_param->post_list + j].send_flags = IBV_SEND_SIGNALED;
-				#ifdef HAVE_IBV_WR_API
```
Comment (dimasique1) [inline src/perftest_resources.c]: I don't understand why did you delete all of that...

Comment (dkkranz) [reply src/perftest_resources.c]: The send flags in post list mode should be determined per WR and not per list,
for example, to set IBV_SEND_SIGNALED according to the cq mod.
The extended QP wr_flags field should be set according to the WR flags before invoking each WR builder.

Comment (dimasique1) [reply src/perftest_resources.c]: Right, okay

**Comment by dimasique1** [comment]
Thanks for the input @dkkranz , LGTM. i need to test yours and Firas's PRs along with others. I also plan to introduce a new branch next week, that one will become master soon. So I will appreciate if you add you PRs for this branch too

**Comment by dkkranz** [comment]
Sure, I will submit another PR when the new branch is out. Is there anything else i can provide in order to merge this one?

**Comment by dkkranz** [comment]
Please wait with merging this PR, I would like to apply a small change

**Comment by dkkranz** [comment]
HAVE_RDMA_READ_SRD is now defined only if WR API was configured, because SRD does not support RDMA read requests using ibv_post_send.

**Comment by dkkranz** [comment]
@dimasique1 This PR is ready to merge

**Comment by galpress** [comment]
> Thanks for the input @dkkranz , LGTM. i need to test yours and Firas's PRs along with others. I also plan to introduce a new branch next week, that one will become master soon. So I will appreciate if you add you PRs for this branch too

@dimasique1 could you please clarify what you mean by that one will become master soon? Are you planning to force push a new branch into master?

**Comment by galpress** [comment]
> I want to do so, not sure how soon this happens

That's not how people usually work with git...
Force pushing is just bad practice, why not merge your existing PR?

File: src/perftest_resources.c [hunk 2]
```diff
@@ -300,71 +300,82 @@ static inline int _new_post_send(struct pingpong_context *ctx,
...
-			ctx->wr[wr_index].qp_type.xrc.remote_srqn);
-	}
-	#endif
+		if (qpt == IBV_QPT_UD) {
+			ibv_wr_set_ud_addr(
+				ctx->qpx[index],
+				wr->wr.ud.ah,
+				wr->wr.ud.remote_qpn,
+				wr->wr.ud.remote_qkey);
+		} else if (qpt == IBV_QPT_DRIVER && user_param->connection_type == SRD) {
```
[the remark this answers is not in the document]
Comment (dkkranz) [reply src/perftest_resources.c]: Done, thanks

**Comment by dkkranz** [comment]
@dimasique1 Please note that i pushed a fix regarding your last comment

**Comment by dimasique1** [comment]
@dkkranz, may I please ask you to port your changes into rdma-core-dc-support branch? I can do it, but I am afraid I cannot test it properly on EFA devices, since I don't have them...

**Comment by galpress** [comment]
> > I want to do so, not sure how soon this happens
> 
> That's not how people usually work with git...
> Force pushing is just bad practice, why not merge your existing PR?

@dimasique1 ?

**Comment by dimasique1** [comment]
> > > I want to do so, not sure how soon this happens
> > 
> > 
> > That's not how people usually work with git...
> > Force pushing is just bad practice, why not merge your existing PR?
> 
> @dimasique1 ?

I didn't force push anything, I merged the existing PR, as you advised.

**Comment by dkkranz** [comment]
> 
> 
> @dkkranz, may I please ask you to port your changes into rdma-core-dc-support branch? I can do it, but I am afraid I cannot test it properly on EFA devices, since I don't have them...

I will try to get to it soon
